custom_components/bond/light.py

```diff
@@ -22,71 +33,151 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     # Setup connection with devices/cloud
     bond = hass.data[DOMAIN]['bond_hub']
 
-    # Verify that passed in configuration works
-    # if not hub.is_valid_login():
-    #     _LOGGER.error("Could not connect to AwesomeLight hub")
-    #     return
-
     # Add devices
     for deviceId in bond.getDeviceIds():
-        newBondLight = BondLight(bond, deviceId)
+        device = self._bond.getDevice(deviceId)
+        deviceType = device['type']
+        actions = device['actions']
+
+        if deviceType == BOND_DEVICE_TYPE_CEILING_FAN:
+            # If the device type is not Ceiling Fan, or it is
+            # Ceiling Fan but has no action for light control
+            # then don't create a light instance.
+            supportsLightActions = \
+                BOND_DEVICE_ACTION_TURN_LIGHT_ON in actions or \
+                BOND_DEVICE_ACTION_TURN_LIGHT_OFF in actions or \
+                BOND_DEVICE_ACTION_TOGGLE_LIGHT in actions
+            if supportsLightActions:
+                deviceProperties = self._bond.getProperties(deviceId)
+                light = BondLight(bond, deviceId, device, deviceProperties)
+                add_entities([light])
```

Comment:
I wonder if it's better to build up a list and call this once

Reply:
Probably. I got that impression, reading the docs when I first wrote this, but this implementation was quicker at the time, and I had no idea what I was doing in Python...

Feel free to implement it as a single call, or we can get it later in another update.

Reply:
I'll pass on this for now, but we should get it in a later update.